### ISSUE TITLE
Migrate maintenance.py to new data model

### ISSUE THREAD

**Issue by re-nobre** [pixelated/pixelated-user-agent#421]
## Note
This story is part of the #420

**Comment by jeffhsta** [comment]
I tried to run maintenance load-mails, I created a new account and run the load-mails I got the error below, and at the end just the mail on INBOX are loaded, and no one for DRAFTS and SENT mail dirs.

2015-08-07 14:40:37 [leap.keymanager.openpgp] INFO About to generate keys... This might take SOME time.
2015-08-07 14:45:25 [gnupg] INFO Key created. Fingerprint: 5A1BFD97266DD6DE5BF85A1C1FA9005AAE828057
2015-08-07 14:45:25 [leap.keymanager.openpgp] INFO Keys for jeff7@unstable.pixelated-project.org have been successfully generated.
2015-08-07 14:45:25 [twisted] INFO Stopping factory <leap.common.http._HTTP11ClientFactory instance at 0x7f3599906dd0>
2015-08-07 14:45:26 [requests.packages.urllib3.connectionpool] INFO Starting new HTTPS connection (1): api.unstable.pixelated-project.org
2015-08-07 14:45:27 [twisted] INFO FETCH: syncing soledad...
2015-08-07 14:45:27 [twisted] INFO Starting factory <leap.common.http._HTTP11ClientFactory instance at 0x7f35998f9560>
2015-08-07 14:45:27 [some logger] WARNING Before sync
2015-08-07 14:45:37 [twisted] INFO FETCH soledad SYNCED.
2015-08-07 14:45:37 [twisted] INFO processing doclist
2015-08-07 14:45:37 [some logger] WARNING after sync
2015-08-07 14:45:37 [twisted] INFO All parts inserted for msg!
2015-08-07 14:45:37 [twisted] INFO All parts inserted for msg!
2015-08-07 14:45:37 [twisted] INFO All parts inserted for msg!
2015-08-07 14:45:37 [twisted] INFO All parts inserted for msg!
Unhandled error in Deferred:
2015-08-07 14:45:37 [twisted] CRITICAL Unhandled error in Deferred:


Traceback (most recent call last):
Failure: twisted.internet.defer.FirstError: FirstError[#6, [Failure instance: Traceback: <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'replace'
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/lib/python2.7/site-packages/twisted/internet/defer.py:393:callback
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/lib/python2.7/site-packages/twisted/internet/defer.py:501:_startRunCallbacks
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/lib/python2.7/site-packages/twisted/internet/defer.py:588:_runCallbacks
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/lib/python2.7/site-packages/twisted/internet/defer.py:1184:gotResult
--- <exception caught here> ---
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/lib/python2.7/site-packages/twisted/internet/defer.py:1128:_inlineCallbacks
/home/jefferson/projects/pixelated-project/pixelated-user-agent/service/pixelated/adapter/mailstore/leap_mailstore.py:135:add_mail
/home/jefferson/projects/pixelated-project/pixelated-user-agent/.venv-0.7.2/src/leap.mail/src/leap/mail/adaptors/soledad.py:649:set_mbox_uuid
]]
2015-08-07 14:45:37 [twisted] CRITICAL 
2015-08-07 14:45:39 [twisted] INFO All parts inserted for msg!
2015-08-07 14:45:40 [twisted] INFO All parts inserted for msg!
2015-08-07 14:46:27 [twisted] INFO FETCH: syncing soledad...
2015-08-07 14:47:14 [twisted] INFO FETCH soledad SYNCED.
2015-08-07 14:47:14 [twisted] INFO processing doclist
2015-08-07 14:47:27 [twisted] INFO FETCH: syncing soledad...
2015-08-07 14:47:28 [twisted] INFO FETCH soledad SYNCED.
2015-08-07 14:47:28 [twisted] INFO processing doclist
2015-08-07 14:48:27 [twisted] INFO FETCH: syncing soledad...
2015-08-07 14:48:28 [twisted] INFO FETCH soledad SYNCED.
2015-08-07 14:48:28 [twisted] INFO processing doclist
2015-08-07 14:49:27 [twisted] INFO FETCH: syncing soledad...
2015-08-07 14:49:28 [twisted] INFO FETCH soledad SYNCED.
2015-08-07 14:49:28 [twisted] INFO processing doclist

**Comment by fbernitt** [comment]
Blocked because develop HEAD of soledad does not work with develop head of leap_pycommon:

2015-08-13 07:26:46 [leap.soledad.client.api] ERROR Soledad exception when syncing!
*--- Failure #6 (pickled) ---
/home/vagrant/pixelated-user-agent-0.7/local/lib/python2.7/site-packages/twisted/internet/defer.py:1274: unwindGenerator(...)
/home/vagrant/pixelated-user-agent-0.7/local/lib/python2.7/site-packages/twisted/internet/defer.py:1128: _inlineCallbacks(...)
/home/vagrant/pixelated-user-agent-0.7/src/leap.soledad.client/client/src/leap/soledad/client/sync.py:73: sync(...)
/home/vagrant/pixelated-user-agent-0.7/local/lib/python2.7/site-packages/twisted/internet/defer.py:1274: unwindGenerator(...)
 [Capture of Locals and Globals disabled (use captureVars=True)]
--- <exception caught here> ---
/home/vagrant/pixelated-user-agent-0.7/local/lib/python2.7/site-packages/twisted/internet/defer.py:1128: _inlineCallbacks(...)
/home/vagrant/pixelated-user-agent-0.7/src/leap.soledad.client/client/src/leap/soledad/client/http_target.py:260: get_sync_info(...)
/home/vagrant/pixelated-user-agent-0.7/src/leap.soledad.client/client/src/leap/soledad/client/http_target.py:687: _http_request(...)
 [Capture of Locals and Globals disabled (use captureVars=True)]
exceptions.TypeError: request() takes at most 5 arguments (6 given)

**Comment by fbernitt** [comment]
Fix is blocked by this pull request: https://github.com/leapcode/leap_pycommon/pull/118

**Comment by kaeff** [comment]
Unblocked after merge of https://github.com/leapcode/leap_pycommon/pull/118

**Comment by kaeff** [comment]
There was a problem due to a method signature mismatch which I fixed in e92431a4d7b71418f354fa9c3a6d466e2d113fa9. 

`reset` is still failing with the following exception and needs fixing:

    2015-08-19 18:50:31 [some logger] WARNING after sync
    Unhandled error in Deferred:
    2015-08-19 18:50:31 [twisted] CRITICAL Unhandled error in Deferred:


    Traceback (most recent call last):
      File "/Users/kfleerko/.virtualenvs/pixelated-user-agent/lib/python2.7/site-packages/twisted/internet/defer.py", line 1184, in gotResult
        _inlineCallbacks(r, g, deferred)
      File "/Users/kfleerko/.virtualenvs/pixelated-user-agent/lib/python2.7/site-packages/twisted/internet/defer.py", line 1171, in _inlineCallbacks
        deferred.callback(e.value)
      File "/Users/kfleerko/.virtualenvs/pixelated-user-agent/lib/python2.7/site-packages/twisted/internet/defer.py", line 393, in callback
        self._startRunCallbacks(result)
      File "/Users/kfleerko/.virtualenvs/pixelated-user-agent/lib/python2.7/site-packages/twisted/internet/defer.py", line 501, in _startRunCallbacks
        self._runCallbacks()
    --- <exception caught here> ---
      File "/Users/kfleerko/.virtualenvs/pixelated-user-agent/lib/python2.7/site-packages/twisted/internet/defer.py", line 588, in _runCallbacks
        current.result = callback(current.result, *args, **kw)
      File "pixelated/maintenance.py", line 103, in delete_all_mails
        generation, docs = soledad.get_all_docs()
    exceptions.TypeError: iteration over non-sequence
    2015-08-19 18:50:31 [twisted] CRITICAL


It seems as whenever an exception occurs during any of the commands, the reactor isn't terminated and the program keeps running. But that is more an annoyance than a problem.

**Comment by kaeff** [comment]
After `6093146c62dfaa300b0c5c7c8fff91cfe25b194e`, all 4 maintenance command are working against unstable. QA ok.
